internal/provider/resource_aci_external_management_network_instance_profile.go

```diff
@@ -0,0 +1,680 @@
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/setplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &MgmtInstPResource{}
+var _ resource.ResourceWithImportState = &MgmtInstPResource{}
+
+func NewMgmtInstPResource() resource.Resource {
+	return &MgmtInstPResource{}
+}
+
+// MgmtInstPResource defines the resource implementation.
+type MgmtInstPResource struct {
+	client *client.Client
+}
+
+// MgmtInstPResourceModel describes the resource data model.
+type MgmtInstPResourceModel struct {
+	Id            types.String `tfsdk:"id"`
+	Annotation    types.String `tfsdk:"annotation"`
+	Descr         types.String `tfsdk:"description"`
+	Name          types.String `tfsdk:"name"`
+	NameAlias     types.String `tfsdk:"name_alias"`
+	Prio          types.String `tfsdk:"priority"`
+	MgmtRsOoBCons types.Set    `tfsdk:"relation_to_consumed_out_of_band_contracts"`
+	TagAnnotation types.Set    `tfsdk:"annotations"`
+}
+
+// MgmtRsOoBConsMgmtInstPResourceModel describes the resource data model for the children without relation ships.
+type MgmtRsOoBConsMgmtInstPResourceModel struct {
+	Annotation      types.String `tfsdk:"annotation"`
+	Prio            types.String `tfsdk:"priority"`
+	TnVzOOBBrCPName types.String `tfsdk:"out_of_band_contract_name"`
+}
+
+// TagAnnotationMgmtInstPResourceModel describes the resource data model for the children without relation ships.
+type TagAnnotationMgmtInstPResourceModel struct {
+	Key   types.String `tfsdk:"key"`
+	Value types.String `tfsdk:"value"`
+}
+
+type MgmtInstPIdentifier struct {
+	Name types.String
+}
+
+func (r *MgmtInstPResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	tflog.Debug(ctx, "Start plan modification of resource: aci_external_management_network_instance_profile")
+
+	if !req.Plan.Raw.IsNull() {
+		var planData *MgmtInstPResourceModel
+		resp.Diagnostics.Append(req.Plan.Get(ctx, &planData)...)
+
+		var MgmtRsOoBConsPlan, MgmtRsOoBConsUpdate []MgmtRsOoBConsResourceModel
+		planData.MgmtRsOoBCons.ElementsAs(ctx, &MgmtRsOoBConsPlan, false)
+		if len(MgmtRsOoBConsPlan) > 0 {
+			for _, MgmtRsOoBCons := range MgmtRsOoBConsPlan {
+				if r.client.ValidateRelationDn {
```

Comment:
add check that DN has been passed by regex with /[\w![!]]*-

Reply:
removed dn input completely after discussion, only name inputs